src/App.tsx

```diff
@@ -1,10 +1,67 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { getTodos } from './api/todos';
+import { ShowState } from './types/ShowState';
+import { ErrorTypes } from './types/ErrorTypes';
+import { Error } from './components/Error';
+import { Loader } from './components/Loader';
+
+const USER_ID = 12177;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [showState, setShowState] = useState<ShowState>(ShowState.All);
+  const [error, setError] = useState<ErrorTypes | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const filteredTodos = useMemo(() => {
+    switch (showState) {
+      case ShowState.All:
+        return todos;
+      case ShowState.Active:
+        return todos.filter(toddo => !toddo.completed);
+      case ShowState.Completed:
+        return todos.filter(toddo => toddo.completed);
+      default:
+        return todos;
+    }
+  }, [showState, todos]);
+
+  const loadAllTodos = async () => {
+    setLoading(true);
+    try {
+      const allTodos = await getTodos(USER_ID);
+
+      return allTodos;
+    } catch (err) {
+      setError(ErrorTypes.LOAD_ALL_TODOS);
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+
+      return undefined;
```

Comment:
redundant

Reply:
When I removed this line, I got an Error.
Expected to return a value at the end of the async arrow function.